querier_test.go

```diff
@@ -1514,11 +1463,13 @@ func (it *listSeriesIterator) Seek(t int64) bool {
 	if it.idx == -1 {
 		it.idx = 0
 	}
+
 	// Do binary search between current position and end.
-	it.idx = sort.Search(len(it.list)-it.idx, func(i int) bool {
```

Comment:
Problem here was that `sort.Search` on not found gives the index to put the value in. This makes sense but seems like we forgot that we operate on shorter `len(it.list)-it.idx` list (: